src/lib.rs

```diff
@@ -90,6 +90,34 @@ fn get_num_cpus() -> usize {
     }
 }
 
+/// Returns the number of physical CPUs of the current machine.
+#[inline]
+pub fn get_physical() -> usize {
+    get_physical_num_cpus()
+}
+
+#[cfg(target_os="macos")]
+fn get_physical_num_cpus() -> usize {
+    use std::ffi::CString;
+    use libc::size_t;
+    use libc::sysctlbyname;
+    use std::ptr;
+    use libc::c_void;
+
+    unsafe {
+        let name = CString::new("hw.physicalcpu").unwrap().as_ptr();
```

Comment:
I can of course do that, but is the performance and/or RAM gain really that big that it outweighs the loss in clarity? I mean, I can't say by looking at the array of numbers what exactly it's supposed to represent.